src/main/webapp/app/course/manage/course-management.service.ts

```diff
@@ -692,4 +702,48 @@ export class CourseManagementService {
         // Note: 0 is the default value in case the server returns something that does not make sense
         return this.http.get<number>(`${this.resourceUrl}/${courseId}/allowed-complaints?teamMode=${teamMode}`) ?? 0;
     }
+
+    enableCourseOverviewBackground() {
+        this.courseOverviewSubject.next(true);
+    }
+
+    disableCourseOverviewBackground() {
+        this.courseOverviewSubject.next(false);
+    }
+
+    /**
+     * Sorts and returns the semesters by year descending
+     * WS is sorted above SS
+     *
+     * @param coursesWithSemesters the courses to sort the semesters of
+     * @return An array of sorted semester names
+     */
+    getUniqueSemesterNamesSorted(coursesWithSemesters: Course[]): string[] {
+        return (
+            coursesWithSemesters
+                // Test courses get their own section later
+                .filter((course) => !course.testCourse)
+                .map((course) => course.semester ?? '')
+                // Filter down to unique values
+                .filter((course, index, courses) => courses.indexOf(course) === index)
+                .sort((semesterA, semesterB) => {
+                    // Sort last if the semester is unset
+                    if (semesterA === '') {
+                        return 1;
+                    }
+                    if (semesterB === '') {
+                        return -1;
+                    }
+
+                    // Parse years in base 10 by extracting the two digits after the WS or SS prefix
+                    const yearsCompared = parseInt(semesterB.slice(2, 4), 10) - parseInt(semesterA.slice(2, 4), 10);
+                    if (yearsCompared !== 0) {
+                        return yearsCompared;
+                    }
+
+                    // If years are the same, sort WS over SS
+                    return semesterA.slice(0, 2) === 'WS' ? -1 : 1;
```

Comment:
Shouldn't you check semesterB here as well? If both are `WS` you should return 0 here.

Reply:
I didn't write that code, just moved that function to the service layer. But I think you are right, it doesn't handle the case when both semesters are the same. I will add this check too 👍